### ISSUE TITLE
Remove 'print friendly' long text of subgroup name, force default name

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/CohortGenerator#135]
The current implementation of a subgroup name from cohort subset definition is to create a long title based on cohortID, the start end windows, and other attributes.   This leads to extremely long titles that actually don't have 'searchable' text.

Example:  you made a subset def to find prior Hypertension within 30d.  That should be the subgruop name: 'prior hypertension within 30d'.  Or even just 'prior hypertension' if the days do not need to be displayed in the name.

There was a feature added to override the default behavior with putting the fully specified subgroup def in the name.  This issue is to remove that behavior (tho, creating a descritpion of the subgroup is something that is useful, just not for use in naming the cohort).  Ie: we should never have the print friendly text used as the default cohort name for the subgroup.

To illustrate this:
![image](https://github.com/OHDSI/CohortGenerator/assets/6818777/96a1b186-c673-47b7-87a0-adad7fdf7104)

The cohort names presented in this plot are not effective.

**Comment by anthonysena** [comment]
Noting that this would be a potentially breaking change which we can plan for v1.0 and then require the use of a subset template name per #118.

**Comment by anthonysena** [comment]
Linking this to #218 which involves having a print friendly way of describing the cohort subsets. Perhaps we can repurpose the current default subset naming to accomplish this task?

**Comment by chrisknoll** [comment]
> Linking this to https://github.com/OHDSI/CohortGenerator/issues/218 which involves having a print friendly way of describing the cohort subsets. Perhaps we can repurpose the current default subset naming to accomplish this task?

Yes.  I don't want this issue to be interpreted as the long descriptive name as something that should be removed.  I would reference this feature as a 'print friendly' type of thing and not a 'default title' sort of thing.  So keep this functionality but expose it in some other way through the API as getting a print-friendly view of the subgroup definition.

**Comment by anthonysena** [comment]
Just to illustrate the naming problem described/discussed in this thread, consider the following example based off of the package vignette where I've commented out the subset operator names and construct 2 subsets: one with the "calculated" name and the other that uses the subset naming template approach:

``` r
jsonFilePath <- system.file("testdata", "CohortsToSubset.JSON", package = "CohortGenerator")
cohortDefinitionSet <- jsonlite::fromJSON(jsonFilePath)
cohortDefinitionSet <- cohortDefinitionSet |>
  dplyr::filter(cohortId == 1)

ageCriteria <- CohortGenerator::createDemographicSubsetOperator(
  ageMin = 20,
  ageMax = 50
)

limitToLastEver <- CohortGenerator::createLimitSubsetOperator(
  #name = "Last event during 1 January 2000 and 31 December 2008",
  priorTime = 0,
  followUpTime = 0,
  limitTo = "lastEver",
  calendarStartDate = as.Date("2000-01-01"),
  calendarEndDate = as.Date("2008-12-31")
)

ibuprofenSubset <- CohortGenerator::createCohortSubsetOperator(
  #name = "ibuprofen exposure",
  cohortIds = 2, # Ibuprofen cohort
  cohortCombinationOperator = "any", # Look for any Ibuprofen exposure
  negate = FALSE, # We want to include (not exclude) participants exposed to Ibuprofen,
  windows = list(
    CohortGenerator::createSubsetCohortWindow(
      startDay = 0,
      endDay = 9999,
      targetAnchor = "cohortStart",
      subsetAnchor = "cohortStart"
    ),
    CohortGenerator::createSubsetCohortWindow(
      startDay = -9999,
      endDay = 0,
      targetAnchor = "cohortEnd",
      subsetAnchor = "cohortStart"
    )
  )
)

ibuprofenWCelcoxib <- CohortGenerator::createCohortSubsetDefinition(
  #name = "Aged 20-50 yrs, last ibuprofen exposure from 2000-2008",
  name = "",
  definitionId = 11, # Unique ID for this subset
  subsetOperators = list(
    ageCriteria,
    ibuprofenSubset,
    limitToLastEver
  )#,
 # subsetCohortNameTemplate = "@baseCohortName - @subsetDefinitionName"
)

ibuprofenWCelcoxibFriendlyNames <- CohortGenerator::createCohortSubsetDefinition(
  name = "Aged 20-50 yrs, last ibuprofen exposure from 2000-2008",
  definitionId = 12, # Unique ID for this subset
  subsetOperators = list(
    ageCriteria,
    ibuprofenSubset,
    limitToLastEver
  ),
  subsetCohortNameTemplate = "@baseCohortName - @subsetDefinitionName"
)


cohortDefinitionSet <- cohortDefinitionSet |>
  CohortGenerator::addCohortSubsetDefinition(ibuprofenWCelcoxib, targetCohortIds = c(1)) |>
  CohortGenerator::addCohortSubsetDefinition(ibuprofenWCelcoxibFriendlyNames, targetCohortIds = c(1))

cohortDefinitionSet |> dplyr::select(cohortId, cohortName)
#>   cohortId
#> 1        1
#> 2     1011
#> 3     1012
#>                                                                                                                                                                                                                                               cohortName
#> 1                                                                                                                                                                                                                                               Celcoxib
#> 2 Celcoxib -  aged 20 - 50, in cohorts: (2) where subset cohort start is within D: 0 - D: 9999 of target cohort start and subset cohort start is within D: -9999 - D: 0 of target cohort end, last ever occurence after 2000-01-01 and before 2008-12-31
#> 3                                                                                                                                                                                      Celcoxib - Aged 20-50 yrs, last ibuprofen exposure from 2000-2008
```

<sup>Created on 2025-09-29 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

In the example above, cohortId == 1011 uses no subset naming and the resulting cohort name is:

_Celcoxib -  aged 20 - 50, in cohorts: (2) where subset cohort start is within D: 0 - D: 9999 of target cohort start and subset cohort start is within D: -9999 - D: 0 of target cohort end, last ever occurence after 2000-01-01 and before 2008-12-31_

While cohortId == 1012 uses the cohort name and templating approach which results in:

_Celcoxib - Aged 20-50 yrs, last ibuprofen exposure from 2000-2008_

So it would seem that a potential approach would be to require a subset definition name and use `subsetCohortNameTemplate = "@baseCohortName - @subsetDefinitionName"` as a default for the `createCohortSubsetDefinition` function. Let me know  your thoughts @chrisknoll @azimov

**Comment by chrisknoll** [comment]
Thanks for the demo @anthonysena .

My personal preference is that you never get the 'print friendly' (aka. the long name) as a cohort name.  If anything, maybe default the name to 'Subset {id}' and they can lookup (via a print friendly operation) what 'Subset {id}' is.  My reason for this is that it will never be the case (IMO) that you will want the cohort name to be as follows:

_Celcoxib - aged 20 - 50, in cohorts: (2) where subset cohort start is within D: 0 - D: 9999 of target cohort start and subset cohort start is within D: -9999 - D: 0 of target cohort end, last ever occurence after 2000-01-01 and before 2008-12-31_

(btw: typo in 'occurence' should be 'occurrence' (2 r's))

It sounds like you're saying @azimov moved the logic to a print friendly class.  That's excellent, and if someone really wants to make the title the print friendly text, make them generate the print friendly text themselves and add it to the template.  We need the print friendly function so that when the user asks 'what is the 'Subgroup A' population? they can get the nice print friendly description of what that means which could include limit, demographic, cohortsubset operators, etc. Let the designer specify how to refer to it in their study with a simple name, and give the users the ability to understand (in plain english) what it means.  This is the way.

TL;DR: I suggest default the subset name to 'subset {id}' as it is specific and short (versus long and unusable), and externalize the function for print friendly so that you can generate a description of the subset definition when the user needs to know it.  Alternatively, force the subset definition name to be provided by the designer.

**Comment by azimov** [comment]
I agree @chrisknoll that we shouldn't display the auto name, its needlessly long and I don't think anyone has ever like it. My feeling is that we should force some sort of naming by the the user, though. We would never allow someone to create a blank cohort definition in atlas, we shouldn't with subsetting

The print friendly is really a separate but related issue.
